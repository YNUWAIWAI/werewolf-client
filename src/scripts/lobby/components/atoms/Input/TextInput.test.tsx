import * as React from 'react'
import {TextInput} from './TextInput'
import {shallow} from 'enzyme'

test('render', () => {
  const handleChangeInner = jest.fn()
  const handleChange = jest.fn(() => handleChangeInner)
  const wrapper = shallow(
    <TextInput
      className="lo--advanced-search--input"
      handleChange={handleChange}
      initialValue=""
      max={30}
      min={5}
      navigatable
      placeholder="5-30 chars"
      required
    />
  )

  expect(wrapper.html()).toMatchSnapshot()
  expect(handleChange).toHaveBeenCalledTimes(0)
})
describe('handleBlur', () => {
  test('valid', () => {
    const handleBlurInner = jest.fn()
    const handleBlur = jest.fn(() => handleBlurInner)
    const wrapper = shallow(
      <TextInput
        className="lo--advanced-search--input"
        handleBlur={handleBlur}
        initialValue=""
        max={30}
        min={5}
        navigatable
        placeholder="5-30 chars"
        required
      />
    )

    wrapper.find('input').simulate('blur', {
      target: {
        validity: {
          valid: true
        },
        value: 'valid text'
      }
    })
    expect(handleBlur).toHaveBeenCalledTimes(1)
    expect(handleBlur).toHaveBeenCalledWith(true)
    expect(handleBlurInner).toHaveBeenCalledTimes(1)
    expect(handleBlurInner).toHaveBeenCalledWith('valid text')
  })
  test('invalid', () => {
    const handleBlurInner = jest.fn()
    const handleBlur = jest.fn(() => handleBlurInner)
    const wrapper = shallow(
      <TextInput
        className="lo--advanced-search--input"
        handleBlur={handleBlur}
        initialValue=""
        max={30}
        min={5}
        navigatable
        placeholder="5-30 chars"
        required
      />
    )

    wrapper.find('input').simulate('blur', {
      target: {
        validity: {
          valid: false
        },
        value: 'text'
      }
    })
    expect(handleBlur).toHaveBeenCalledTimes(1)
    expect(handleBlur).toHaveBeenCalledWith(false)
    expect(handleBlurInner).toHaveBeenCalledTimes(1)
    expect(handleBlurInner).toHaveBeenCalledWith('text')
  })
})
describe('handleChange', () => {
  test('valid', () => {
    const handleChangeInner = jest.fn()
    const handleChange = jest.fn(() => handleChangeInner)
    const wrapper = shallow(
      <TextInput
        className="lo--advanced-search--input"
        handleChange={handleChange}
        initialValue=""
        max={30}
        min={5}
        navigatable
        placeholder="5-30 chars"
        required
      />
    )

    wrapper.find('input').simulate('change', {
      target: {
        validity: {
          valid: true
        },
        value: 'valid text'
      }
    })
    expect(handleChange).toHaveBeenCalledTimes(1)
    expect(handleChange).toHaveBeenCalledWith(true)
    expect(handleChangeInner).toHaveBeenCalledTimes(1)
    expect(handleChangeInner).toHaveBeenCalledWith('valid text')
  })
  test('invalid', () => {
    const handleChangeInner = jest.fn()
    const handleChange = jest.fn(() => handleChangeInner)
    const wrapper = shallow(
      <TextInput
        className="lo--advanced-search--input"
        handleChange={handleChange}
        initialValue=""
        max={30}
        min={5}
        navigatable
        placeholder="5-30 chars"
        required
      />
    )

    wrapper.find('input').simulate('change', {
      target: {
        validity: {
          valid: false
        },
        value: 'text'
      }
    })
    expect(handleChange).toHaveBeenCalledTimes(1)
    expect(handleChange).toHaveBeenCalledWith(false)
    expect(handleChangeInner).toHaveBeenCalledTimes(1)
    expect(handleChangeInner).toHaveBeenCalledWith('text')
  })
})
