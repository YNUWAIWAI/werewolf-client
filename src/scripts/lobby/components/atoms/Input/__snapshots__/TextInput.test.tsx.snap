// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `"<input type=\\"text\\" class=\\"lo--advanced-search--input\\" maxLength=\\"30\\" minLength=\\"5\\" placeholder=\\"5-30 chars\\" required=\\"\\" tabindex=\\"0\\" value=\\"\\"/>"`;
